### PR TITLE
12.0 mig product average consumption

### PR DESCRIPTION
**Pull request by ivantodorovich**

### CODE REVIEW

**Comment by ivantodorovich** [comment]
@krunal267 please check travis errors

PEP8
```
product_average_consumption/models/res_config.py:65:9: E265 block comment should start with '# '
product_average_consumption/models/product_product.py:37:80: E501 line too long (100 > 79 characters)
product_average_consumption/models/product_product.py:38:80: E501 line too long (120 > 79 characters)
product_average_consumption/models/product_product.py:39:80: E501 line too long (96 > 79 characters)
product_average_consumption/models/product_product.py:40:80: E501 line too long (105 > 79 characters)
product_average_consumption/models/product_product.py:41:9: E128 continuation line under-indented for visual indent
product_average_consumption/models/product_product.py:42:9: E128 continuation line under-indented for visual indent
product_average_consumption/models/product_product.py:43:9: E128 continuation line under-indented for visual indent
product_average_consumption/models/product_product.py:44:80: E501 line too long (111 > 79 characters)
product_average_consumption/models/product_product.py:46:80: E501 line too long (83 > 79 characters)
product_average_consumption/models/product_product.py:56:80: E501 line too long (84 > 79 characters)
product_average_consumption/models/product_product.py:81:80: E501 line too long (91 > 79 characters)
product_average_consumption/models/product_product.py:85:80: E501 line too long (98 > 79 characters)
product_average_consumption/models/product_product.py:87:80: E501 line too long (92 > 79 characters)
product_average_consumption/models/product_product.py:88:32: E127 continuation line over-indented for visual indent
product_average_consumption/models/product_product.py:88:80: E501 line too long (82 > 79 characters)
product_average_consumption/models/product_product.py:89:80: E501 line too long (121 > 79 characters)
product_average_consumption/models/product_product.py:90:80: E501 line too long (89 > 79 characters)
product_average_consumption/models/product_product.py:91:80: E501 line too long (114 > 79 characters)
product_average_consumption/models/product_product.py:92:80: E501 line too long (107 > 79 characters)
product_average_consumption/models/product_product.py:93:80: E501 line too long (89 > 79 characters)
product_average_consumption/models/product_product.py:101:80: E501 line too long (103 > 79 characters)
product_average_consumption/models/product_template.py:39:80: E501 line too long (108 > 79 characters)
product_average_consumption/models/product_template.py:40:80: E501 line too long (120 > 79 characters)
product_average_consumption/models/product_template.py:41:80: E501 line too long (104 > 79 characters)
product_average_consumption/models/product_template.py:42:80: E501 line too long (108 > 79 characters)
product_average_consumption/models/product_template.py:43:33: E127 continuation line over-indented for visual indent
product_average_consumption/models/product_template.py:43:80: E501 line too long (96 > 79 characters)
product_average_consumption/models/product_template.py:44:33: E127 continuation line over-indented for visual indent
product_average_consumption/models/product_template.py:44:80: E501 line too long (88 > 79 characters)
product_average_consumption/models/product_template.py:46:80: E501 line too long (90 > 79 characters)
product_average_consumption/models/product_template.py:47:80: E501 line too long (104 > 79 characters)
product_average_consumption/models/product_template.py:48:36: E201 whitespace after '('
product_average_consumption/models/product_template.py:52:80: E501 line too long (153 > 79 characters)
product_average_consumption/models/product_template.py:53:9: E128 continuation line under-indented for visual indent
product_average_consumption/models/product_template.py:54:9: E128 continuation line under-indented for visual indent
product_average_consumption/models/product_template.py:57:80: E501 line too long (111 > 79 characters)
product_average_consumption/models/product_template.py:70:80: E501 line too long (97 > 79 characters)
product_average_consumption/models/product_template.py:73:80: E501 line too long (99 > 79 characters)
product_average_consumption/models/product_template.py:79:80: E501 line too long (106 > 79 characters)
product_average_consumption/tests/test_product_average_conumption.py:18:36: W292 no newline at end of file
```

```
************* Module product_average_consumption
product_average_consumption/__init__.py:1: [C7902(missing-readme), ] Missing ./README.rst file. Template here: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
product_average_consumption/views/res_config_view.xml:1: [W7908(missing-newline-extrafiles), ]  Missing newline
************* Module product_average_consumption.__manifest__
product_average_consumption/__manifest__.py:26: [C8103(manifest-deprecated-key), ] Deprecated key "description" in manifest file
************* Module product_average_consumption.models.res_config
product_average_consumption/models/res_config.py:56: [E8103(sql-injection), StockConfigSettings.create] SQL injection risk. Use parameters if you can. - More info https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#no-sql-injection
product_average_consumption/models/res_config.py:61: [E8103(sql-injection), StockConfigSettings.create] SQL injection risk. Use parameters if you can. - More info https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#no-sql-injection
product_average_consumption/models/res_config.py:72: [E8103(sql-injection), StockConfigSettings.create] SQL injection risk. Use parameters if you can. - More info https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#no-sql-injection
product_average_consumption/models/res_config.py:81: [E8103(sql-injection), StockConfigSettings.write] SQL injection risk. Use parameters if you can. - More info https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#no-sql-injection
product_average_consumption/models/res_config.py:86: [E8103(sql-injection), StockConfigSettings.write] SQL injection risk. Use parameters if you can. - More info https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#no-sql-injection
product_average_consumption/models/res_config.py:97: [E8103(sql-injection), StockConfigSettings.write] SQL injection risk. Use parameters if you can. - More info https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#no-sql-injection
product_average_consumption/models/res_config.py:107: [E8103(sql-injection), StockConfigSettings._onchange_default_calculation_range] SQL injection risk. Use parameters if you can. - More info https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#no-sql-injection
************* Module product_average_consumption.models.product_product
product_average_consumption/models/product_product.py:37: [C8108(method-compute), ProductProduct] Name of compute method should start with "_compute_"
product_average_consumption/models/product_product.py:38: [C8108(method-compute), ProductProduct] Name of compute method should start with "_compute_"
product_average_consumption/models/product_product.py:39: [C8108(method-compute), ProductProduct] Name of compute method should start with "_compute_"
product_average_consumption/models/product_product.py:40: [C8108(method-compute), ProductProduct] Name of compute method should start with "_compute_"
product_average_consumption/models/product_product.py:51: [E8103(sql-injection), ProductProduct._min_date] SQL injection risk. Use parameters if you can. - More info https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#no-sql-injection
************* Module product_average_consumption.models.product_template
product_average_consumption/models/product_template.py:40: [C8108(method-compute), ProductTemplate] Name of compute method should start with "_compute_"
```

**Comment by ivantodorovich** [comment]
We should definitely improve the sql injection risk:
https://github.com/druidoo/FoodCoops/pull/5/files#diff-ca25149d0ddd7c594a753a1ce45d427bR59
Anyone with access to the RPC can inject sql by calling create.

Check https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#no-sql-injection, it explains how to safely execute sql statements

File: product_average_consumption/models/product_product.py [hunk 1]
```diff
@@ -23,36 +26,24 @@
...
-        compute='_average_consumption',
-        string='Total Consumption', multi='average_consumption')
-    nb_days = fields.Integer(
-        compute='_average_consumption',
-        string='Number of days for the calculation',
-        multi='average_consumption',
+    average_consumption = fields.Float(compute='_average_consumption', string='Average Consumption')
+    displayed_average_consumption = fields.Float(compute='_displayed_average_consumption', string='Average Consumption')
+    total_consumption = fields.Float(compute='_average_consumption', string='Total Consumption')
+    nb_days = fields.Integer(compute='_average_consumption', string='Number of days for the calculation',
```
Comment (ivantodorovich) [inline product_average_consumption/models/product_product.py]: This is actually breaking PEP8

You can try to define fields like this instead:
```py
average_consumption = fields.Float(
    compute='_average_consumption',
    string='Average Consumption',
)
displayed_average_consumption = fields.Float(
    compute='_displayed_average_consumption',
    string='Average Consumption',
)
```

1 attribute per line
always end with comma

this results in smaller diffs when adding/removing attributes

File: product_average_consumption/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    * Copyright (C) 2019-Today: La Louve (<https://cooplalouve.fr>)
+    * Copyright (C) 2019-Today: Druidoo (<https://www.druidoo.io>)
+    * Copyright : 2013-Today, Groupement Régional Alimentaire de Proximité;
+    * Author :
+        * Druidoo
+        * Julien WESTE;
+        * Sylvain LE GAL (https://twitter.com/legalsylvain);
+    * Licence : AGPL-3 (http://www.gnu.org/licenses/)
+    """,
+    'author': 'Druidoo',
```
Comment (ivantodorovich) [inline product_average_consumption/__manifest__.py]: Sorry for not saying it before, but here we've decided to keep the original authors, and add Druidoo like this:

```
'author': 'Original Author, Druidoo'
```

```
'author': 'Author 1, Author 2, Author 3, Druidoo'
```

File: product_average_consumption/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    'website': 'https://www.druidoo.io',
```
Comment (ivantodorovich) [inline product_average_consumption/__manifest__.py]: Sorry for this too, we also decided to use this website on all modules:
`http://cooplalouve.fr/`

File: product_average_consumption/models/product_product.py (hunk 1 above)
Comment (krunal267) [reply product_average_consumption/models/product_product.py]: Okay, will do it

File: product_average_consumption/__manifest__.py (hunk 2 above)
Comment (krunal267) [reply product_average_consumption/__manifest__.py]: okay, noted.

File: product_average_consumption/__manifest__.py (hunk 3 above)
Comment (krunal267) [reply product_average_consumption/__manifest__.py]: ok, is it http or https?
while I open **http**://cooplalouve.fr, it automatically redirects to **https**://cooplalouve.fr/.

**Comment by ivantodorovich** [comment]
Great! It looks good!

Code coverage is low though. But I guess we can add more tests later. If you can add some now that's be great though
Also if you can squash the migration commits, so we keep it clean before merging 😄

**Comment by ivantodorovich** [comment]
@krunal267 can you check this?

> Great! It looks good!
>
>Code coverage is low though. But I guess we can add more tests later. If you can add some now that's be great though
>Also if you can squash the migration commits, so we keep it clean before merging smile

**Pull request merged**